cmd/brew-report-issue.rb

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env ruby
+# Creates and closes failure debugging issues on a project.
+close = !!ARGV.delete("--close")
+user_repo = ARGV.shift
+message = ARGV.shift
+
+if user_repo.to_s.empty? || message.to_s.empty?
+  abort "Usage: brew report-issue [--close] <user/repo> <message> [<STDIN piped body>]"
+end
+
+unless close
+  abort "Error: the issue/comment body should be piped over STDIN!" if STDIN.tty?
+  body = STDIN.read
+end
+
+github_credentials=`printf "protocol=https\nhost=github.com\n" | git credential fill 2>/dev/null`
+/username=(?<github_username>.+)/ =~ github_credentials
+/password=(?<github_password>.+)/ =~ github_credentials
+github_username ||= ENV["BOXEN_GITHUB_LOGIN"]
+github_username ||= `git config github.user`.chomp
+
+strap_url = ENV["STRAP_URL"]
+strap_url ||= "https://strap.githubapp.com"
+
+if github_username.to_s.empty?
+  abort <<-EOS
+Error: your GitHub username is not set! Set it by running Strap:
+  #{strap_url}
+EOS
+end
+@github_username = github_username
+
+if github_password.to_s.empty?
+  abort <<-EOS
+Error: your GitHub password is not set! Set it by running Strap:
+  #{strap_url}
+EOS
+end
+@github_password = github_password
+
+require "net/http"
+require "json"
+
+def http_request type, url, body=nil
+  uri = URI url
+  request = if type == :post
+    post_request = Net::HTTP::Post.new uri
+    post_request.body = body
+    post_request
+  elsif type == :get
+    Net::HTTP::Get.new uri
+  end
+  return unless request
+  request.basic_auth @github_username, @github_password
+  Net::HTTP.start uri.hostname, uri.port, use_ssl: true  do |http|
+    http.request request
+  end
+end
+
+def response_check response, action
+  return if response.is_a? Net::HTTPSuccess
+  STDERR.puts "Error: failed to #{action}!"
+  unless response.body.empty?
+    failure = JSON.parse response.body
+    STDERR.puts "--\n#{response.code}: #{failure["message"] }"
+  end
+  exit 1
+end
+
+def json_escape string
+  JSON.generate string, quirks_mode: true
+end
+
+def create_issue user_repo, title, body
+  title_json_string = json_escape title
+  body_json_string = json_escape body.chomp
+  new_issue_json = <<-EOS
+    {
+      "title": #{title_json_string},
+      "body":  #{body_json_string}
+    }
+  EOS
+  issues_url = "https://api.github.com/repos/#{user_repo}/issues"
+  response = http_request :post, issues_url, new_issue_json
+  response_check response, "create issue (#{issues_url})"
+  issue = JSON.parse response.body
+  puts "Created issue: #{issue["html_url"]}"
+  issue
+end
+
+def comment_issue issue, comment_body, options={}
+  comments_url = issue["comments_url"]
+  body_json_string = json_escape comment_body.chomp
+  issue_comment_json = "{ \"body\": #{body_json_string} }"
```

Comment:
I'm not sure if it's an improvement, but could also `{:body => comment_body.chomp}.to_json` since `require "json"` happened by this point. Same in the other json spots.